ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -1161,6 +1164,33 @@ ORK_CLASS_AVAILABLE
 @interface ORKTextAnswerFormat : ORKAnswerFormat
 
 /**
+ Returns an initialized text answer format using the regular expression.
+ 
+ This method is one of the designated initializers.
+ 
+ @param expression                The regular expression used to valide the text.
+ @param description               The description of the valid input presented to the user upon invalid input.
+ 
+ @return An initialized validated text answer format.
+ */
+- (instancetype)initWithValidationExpression:(nullable NSString *)expression
+                       validInputDescription:(nullable NSString *)description NS_DESIGNATED_INITIALIZER;
```

Comment:
Both parameters should not be `nullable` so that this way we can enforce that this initializer doesn't get used if they just mean to `init`. It also enforces that a invalid message is provided if a validation regex is applied.